Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -11,10 +11,135 @@
 #include <AMReX_MultiFab.H>
 
 
+/**
+* \brief Get the value of arr in the neighbor (i_n, j_n) on the plane with normal 'dim'.
+*        I.E. If dim==0 it return arr(i, j + i_n, k + j_n),
+*             if dim==1 it return arr(i + i_n, j, k + j_n),
+*             if dim==2 it return arr(i + i_n, j + j_n, k)
+*
+* \param[in] arr \c Array4 To be accessed
+* \param[in] i, j, k the indices of the "center" cell
+* \param[in] i_n the offset of the neighbor in the first direction
+* \param[in] j_n the offset of the neighbor in the second direction
+* \param[in] dim normal direction to the plane in consideration (0 for x, 1 for y, 2 for z)
+*/
+template <class T>
+AMREX_GPU_DEVICE
+constexpr
+T
+GetNeigh(const amrex::Array4<T>& arr,
+                const int i, const int j, const int k,
+                const int i_n, const int j_n, const int dim){
+
+    if(dim == 0){
+        return arr(i, j + i_n, k + j_n);
+    }else if(dim == 1){
+#ifdef WARPX_DIM_XZ
+        return arr(i + i_n, j + j_n, k);
+#elif defined(WARPX_DIM_3D)
+        return arr(i + i_n, j, k + j_n);
+#else
+        amrex::Abort("GetNeigh: Only implemented in 2D3V and 3D3V");
+#endif
+    }else if(dim == 2){
+        return arr(i + i_n, j + j_n, k);
+    }
+
+    amrex::Abort("GetNeigh: dim must be 0, 1 or 2");
+
+    return -1;
+}
+
+
+/**
+* \brief Set the value of arr in the neighbor (i_n, j_n) on the plane with normal 'dim'.
+*        I.E. If dim==0 it return arr(i, j + i_n, k + j_n),
+*             if dim==1 it return arr(i + i_n, j, k + j_n),
+*             if dim==2 it return arr(i + i_n, j + j_n, k)
+*
+* \param[in] arr \c Array4 to be modified
+* \param[in] val the value to be set
+* \param[in] i, j, k the indices of the "center" cell
+* \param[in] i_n the offset of the neighbor in the first direction
+* \param[in] j_n the offset of the neighbor in the second direction
+* \param[in] dim normal direction to the plane in consideration (0 for x, 1 for y, 2 for z)
+*/
+template <class T>
+AMREX_GPU_DEVICE
+constexpr
+void
+SetNeigh(const amrex::Array4<T>& arr, const T val,
+                const int i, const int j, const int k,
+                const int i_n, const int j_n, const int dim){
```

Comment:
Just some alignment (:
```suggestion
SetNeigh(const amrex::Array4<T>& arr, const T val,
         const int i, const int j, const int k,
         const int i_n, const int j_n, const int dim){
```